include/pybind11/pybind11.h

```diff
@@ -458,7 +459,16 @@ class cpp_function : public function {
 
         rec->signature = guarded_strdup(signature.c_str());
         rec->args.shrink_to_fit();
+        for (size_t i = 0; i < rec->args.size(); i++) {
```

Comment:
There is a lowbound of argument_index.size() at least. Argument_index reserve method should be called to prevent many reallocations.